NOCList-Obj-C/NOCList-Obj-C/main.m

```diff
@@ -0,0 +1,113 @@
+
+        LSIAgent *franz = [[LSIAgent alloc] initWithCoverName:@"Franz Krieger" realName:@"Jean Reno" accessLevel:4 compromised:NO];
+
+        LSIAgent *luther = [[LSIAgent alloc] initWithCoverName:@"Luther Stickell" realName:@"Ving Rhames" accessLevel:4 compromised:NO];
+
+        LSIAgent *sarah = [[LSIAgent alloc] initWithCoverName:@"Sarah Davies" realName:@"Kristen Scott Thomas" accessLevel:5 compromised:YES];
+
+        LSIAgent *max = [[LSIAgent alloc] initWithCoverName:@"Max RotGrab" realName:@"Vanessa Redgrave" accessLevel:4 compromised:NO];
+
+        LSIAgent *hannah = [[LSIAgent alloc] initWithCoverName:@"Hannah Williams" realName:@"Ingeborga Dapkūnaitė" accessLevel:5 compromised:YES];
+
+        LSIAgent *jack = [[LSIAgent alloc] initWithCoverName:@"Jack Harmon" realName:@"Emilio Estevez" accessLevel:6 compromised:YES];
+
+        LSIAgent *frank = [[LSIAgent alloc] initWithCoverName:@"Frank Barnes" realName:@"Dale Dye" accessLevel:9 compromised:NO];
+
+        NSArray *agents = [[NSArray alloc] initWithObjects:ethan,
+                           jim,
+                           claire,
+                           eugene,
+                           franz,
+                           luther,
+                           sarah,
+                           max,
+                           hannah,
+                           jack,
+                           frank,
+                           nil];
+
+        NSMutableArray<LSIAgent *> *compromisedAgents = [[NSMutableArray alloc] init];
+
+        int compromisedAgentCount = 0;
+        for (LSIAgent *agent in agents) {
+            if (agent.compromised.boolValue) {
+                compromisedAgentCount ++;
+                [compromisedAgents addObject:agent];
+            }
+        }
+        for (LSIAgent *agent in compromisedAgents) {
+            NSLog(@"%@ is compromised", agent.coverName);
+        }
+
+        NSLog(@"There are %i compromised agents.", compromisedAgentCount);
+
+        int agentCount = 0;
+        for (LSIAgent *agent in agents) {
+            if (agent.compromised.boolValue == NO) {
+                agentCount ++;
+                NSLog(@"Agent is clean: %@", agent.coverName);
+            }
+        }
+        NSLog(@"There is %i clean agents", agentCount);
+
+        int compAgentCount = 0;
+        for (LSIAgent *agent in agents) {
+            if (agent.compromised.boolValue) {
+                compAgentCount ++;
+
+            }
+        }
+
+        for (LSIAgent *agent in agents) {
+            if (agent.accessLevel.intValue >= 8) {
+                if (agent.compromised.boolValue) {
+                    NSLog(@"Agent %@ with access level %@ is considered high risk. **WARNING** **COMPROMISED**", agent.realName, agent.accessLevel);
+                } else {
+                    NSLog(@"Agent %@ with access level %@ is considered high risk.", agent.realName, agent.accessLevel);
+                }
```

Comment:
This definitely works great! You might consider how you could accomplish this using just one `NSLog` rather than copy/pasting the code (definitely not required; I just am always looking for ways to not repeat myself 😅).